modules/rpc/handler.go

```diff
@@ -97,3 +98,41 @@ func (f fxContextOnewayInboundMiddleware) HandleOneway(ctx context.Context, req
 	fxctx := fxcontext.New(ctx, f.Host)
 	return handler.HandleOneway(fxctx, req)
 }
+
+type authInboundMiddleware struct {
+	service.Host
+}
+
+func (a authInboundMiddleware) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, handler transport.UnaryHandler) error {
+	fxctx, err := authorize(ctx, a.Host)
+	if err != nil {
+		return err
```

Comment:
This is going to surface as a server-side error... we should define a first-class error in YARPC so that we can handle AuthZ errors seperately from server error.

Created https://github.com/yarpc/yarpc-go/issues/653 - you should probably put a TODO here for now.